src/tests/universal-nrp-e2e/README.md

```diff
@@ -0,0 +1,265 @@
+# Universal NRP End-to-End Tests Local/GitHub
+This directory contains tests for the universal NRP both for local/onprem and GitHub workflows. There is a GitHub workflow which performs these tests but only for modern distros (excluding EOL distributions eg. centos 7-8, ubuntu 18.04-20.04, etc.). We do support EOL testing on our infrastructure, as such, we've developed some tools in order for us to test locally.
+
+The following bash tools have been created to enable testing across any linux-based OS:
+ - `StartLocalTest.sh`: Peforms tests locally with a given Azure Policy package. This is meant to be used dirtectly on the machine being tested.
+ - `StartVMTest.sh`: Uses QEMU with a given cloud based image (cloud-init enabled) and performs the tests on the VM.
+ - `StartMSFTTests.sh` (MSFT INTERNAL-ONLY): Uses the _StartVMTest.sh_ above to orchestrate tests on images not being tested by GitrHub workflows. These are internal as they leverage Azure Blob Storage for storing the disk images.
+
+# Testing on VMs
+To perform tests on VMs the `StartVMTest.sh` allows you to perform tests on specific Azure Policy packages targetting a particular VM disk image. QEMU is used as the virtualization platform and allows us to target linux distributions shared as raw/qcow2 images (See [VM Image Sources](#vm-image-sources) for compatible images). Images are booted using cloud-init in order to provision the user and help orchestrate tests, collect logs and test reports back to the QEMU host.
+
+## Example
+The following example performs tests on an [Ubuntu Noble cloud image](https://cloud-images.ubuntu.com/noble/current/) on the AzureLinuxBaseline.zip (built in directory tree) which has 168 resources defined.
+```sh
+# Download image https://cloud-images.ubuntu.com/
+wget https://cloud-images.ubuntu.com/noble/current/noble-server-cloudimg-amd64.img
+# Add 2Gb to filesystem in order to install all necessary dependencies
+qemu-img resize noble-server-cloudimg-amd64.img +2G
+./StartVMTest.sh -i noble-server-cloudimg-amd64.img -p ../../build/AzureLinuxBaseline.zip -c 168
+```
+Once completed, log archives are created under `_<distro-image>` directory (in this case __focal-server-cloudimg-amd64.img_) which contain the JUnit test report along with the osconfig logs contained under `/var/log/osconfig*`
+
+# Testing directly on target machine
+To perform tests directly on a target machine the `StartLocalTest.sh` performs the tests locally.
+
+You need to copy over the following files onto the target machine:
+ - UniversalNRP.Tests.ps1
+ - StartLocalTest.sh
+ - A policy package to apply on the machine - for this example, lets assume `AzureLinuxBaseline.zip`
+
+## Example
+```sh
+./StartLocalTest.sh -p AzureLinuxBaseline.zip -c 168
+```
+
+# Advanced Topics
+## Testing Locally
```

Comment:
Maybe add for these advanced topics that most users likely won't need them?

Reply:
Not sure i understand? These are under the `Advanced Topics` already?

Reply:
Sorry for any misunderstanding. Meant to add a note stating that most users won't need these advanced topics. As it is stated now, it kind of suggests that advanced topics are the way to do local testing.